Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -2609,6 +2614,8 @@ void MediaPlayerPrivateGStreamer::updateStates()
                 m_cachedPosition = MediaTime::invalidTime();
                 GST_DEBUG_OBJECT(pipeline(), "[Seek] seeking to %s failed", toString(m_seekTime).utf8().data());
             }
+        } else if (m_isSeeking && !(state == GST_STATE_PLAYING && pending == GST_STATE_PAUSED)) {
```

Comment:
Can you describe the real life situation which triggers the need for these non-async seeks? My understanding, [as per the gst docs](https://gstreamer.freedesktop.org/documentation/gstreamer/gstelement.html?gi-language=c#gst_element_seek), is that all the seeks should be async except for non-flushing seeks, but those shouldn't be happening in the player private.

Reply:
Well, in comcast we have a device where we have an audio sink which does not support async state changes i.e. it completes transition to PAUSED state on preroll. The video sync on that device works as expected, so in case of videos everything works as expected. The problem, however, arises in audio-only cases.